db/jdbc/src/main/java/io/helidon/db/jdbc/JdbcStatementGet.java

```diff
@@ -0,0 +1,146 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.db.jdbc;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.helidon.common.mapper.MapperManager;
+import io.helidon.common.reactive.Flow;
+import io.helidon.db.DbException;
+import io.helidon.db.DbMapperManager;
+import io.helidon.db.DbRow;
+import io.helidon.db.DbStatement;
+import io.helidon.db.DbStatementType;
+import io.helidon.db.common.InterceptorSupport;
+
+/**
+ * A JDBC get implementation.
+ */
+class JdbcStatementGet implements DbStatement<JdbcStatementGet, CompletionStage<Optional<DbRow>>> {
+    private final JdbcStatementQuery query;
+
+    JdbcStatementGet(ConnectionPool connectionPool,
+                     ExecutorService executorService,
+                     String statementName,
+                     String statement,
+                     DbMapperManager dbMapperMananger,
+                     MapperManager mapperManager,
+                     InterceptorSupport interceptors) {
+
+        this.query = new JdbcStatementQuery(DbStatementType.GET,
+                                            connectionPool,
+                                            executorService,
+                                            statementName,
+                                            statement,
+                                            dbMapperMananger,
+                                            mapperManager,
+                                            interceptors);
+    }
+
+    @Override
+    public JdbcStatementGet params(List<?> parameters) {
+        query.params(parameters);
+        return this;
+    }
+
+    @Override
+    public JdbcStatementGet params(Map<String, ?> parameters) {
+        query.params(parameters);
+        return this;
+    }
+
+    @Override
+    public <T> JdbcStatementGet namedParam(T parameters) {
+        query.namedParam(parameters);
+        return this;
+    }
+
+    @Override
+    public <T> JdbcStatementGet indexedParam(T parameters) {
+        query.indexedParam(parameters);
+        return this;
+    }
+
+    @Override
+    public JdbcStatementGet addParam(Object parameter) {
+        query.addParam(parameter);
+        return this;
+    }
+
+    @Override
+    public JdbcStatementGet addParam(String name, Object parameter) {
+        query.addParam(name, parameter);
+        return this;
+    }
+
+    @Override
+    public CompletionStage<Optional<DbRow>> execute() {
+        CompletableFuture<Optional<DbRow>> result = new CompletableFuture<>();
+
+        query.execute()
+                .publisher()
+                .subscribe(new Flow.Subscriber<DbRow>() {
```

Comment:
I would like to move those anonymou s Subscriber instances into a separate classes. It would make code more readable.